### PR TITLE
Region containment query utils

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

Based on #1770.

This PR introduces utils for querying region containment of objects via point containment checks on bounding box global keypoints.

Supporting utils enable bounding box and keypoint extraction for ArticulatedObjects and ArticulatedLinks.

## How Has This Been Tested

Added pytest using hab3_benchmark HSSD scenes.
Added region annotations to one benchmark scene and updated CI to gather those assets.

TODO: follow-up should add some unit tests for bounding box and keypoint utils tested indirectly here through the query API.

## Types of changes

- **\[Development\]** A pull request that add new features to the [habitat-lab](/habitat-lab) task and environment codebase. Development Pull Requests must be small (less that 500 lines of code change), have unit testing, very extensive documentation and examples. These are typically new tasks, environments, sensors, etc... The review process for these Pull Request is longer because these changes will be maintained by our core team of developers, so make sure your changes are easy to understand!

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes if required.

### CODE REVIEW

File: habitat-lab/habitat/sims/habitat_simulator/sim_utilities.py [hunk 1]
```diff
@@ -450,13 +530,79 @@ def get_rigid_object_global_keypoints(
...
+    :param object_id: The integer id for the object from which to extract keypoints.
+    :param ao_link_map: A pre-computed map from link object ids to their parent ArticulatedObject's object id. If not provided, recomputed as necessary.
+    :param ao_aabbs: A pre-computed map from ArticulatedObject object_ids to their local bounding boxes. If not provided, recomputed as necessary.
+
+    :return: A set of global 3D keypoints for the object.
+    """
+
+    obj = get_obj_from_id(sim, object_id, ao_link_map)
+
+    if type(obj) == habitat_sim.physics.ManagedBulletRigidObject:
```
Comment (jacobkrantz) [inline habitat-lab/habitat/sims/habitat_simulator/sim_utilities.py:596]: I think this should use an "is" relation, or better yet isinstance() for speed

File: habitat-lab/habitat/sims/habitat_simulator/sim_utilities.py [hunk 2]
```diff
@@ -605,3 +751,76 @@ def within(
...
+
+
+def object_in_region(
+    sim: habitat_sim.Simulator,
+    objectA: Union[
+        habitat_sim.physics.ManagedRigidObject,
+        habitat_sim.physics.ManagedArticulatedObject,
+    ],
+    region: habitat_sim.scene.SemanticRegion,
+    containment_threshold=0.25,
```
Comment (jacobkrantz) [inline habitat-lab/habitat/sims/habitat_simulator/sim_utilities.py:763]: Inconsistent use of keypoint thresholding across predicates; `within` uses  `keypoint_vote_threshold` to the same effect.

**Review by jacobkrantz — APPROVED** [review]
nits

File: habitat-lab/habitat/sims/habitat_simulator/sim_utilities.py (hunk 2 above)
Comment (aclegg3) [reply habitat-lab/habitat/sims/habitat_simulator/sim_utilities.py:763]: Thanks, great point.
I agree that "within" and "object_in_region" are using similar but distinct mechanisms for mostly the same purpose. They could be unified.
For now, merging as-is and we'll decide if a follow-up refactor is desired.

**Pull request merged**